deepreg/model/loss/image.py

```diff
@@ -82,3 +84,43 @@ def ssd(y_true: tf.Tensor, y_pred: tf.Tensor) -> tf.Tensor:
     :return: shape = (batch,)
     """
     return tf.reduce_mean(tf.square(y_true - y_pred), axis=[1, 2, 3, 4])
+
+
+def global_mutual_information(y_true: tf.Tensor, y_pred: tf.Tensor) -> tf.Tensor:
+    """
+    differentiable global mutual information loss via Parzen windowing method.
+    reference: https://dspace.mit.edu/handle/1721.1/123142
```

Comment:
this reference is relatively long, is there any specific equation we can use as a reference?

Reply:
also i found the code is a bit similar compared to https://github.com/voxelmorph/voxelmorph/blob/legacy/src/losses.py#L352

if we are referring to some code, maybe we should put a link in docstring?
(like if there's a bug later, it's because of the source lol)

Reply:
btw, just to be fair, the lncc loss I wrote earlier was not a good example neither lol
will improve it

Reply:
As the above one.

Reply:
> 
> 
> also i found the code is a bit similar compared to https://github.com/voxelmorph/voxelmorph/blob/legacy/src/losses.py#L352
> 
> if we are referring to some code, maybe we should put a link in docstring?
> (like if there's a bug later, it's because of the source lol)

sure, the method is from the same paper and I use their code as a reference. Will add the link if necessary.

Reply:
@QianyeYang not necessary if you didn't use them directly lol Im saying this just bcz sometimes I will take a look as existing code and make a comparison to validate my code ;)